lib/symbol-provider.coffee

```diff
@@ -83,20 +88,70 @@ class SymbolProvider
     # Should we disqualify TextEditors with the Grammar text.plain.null-grammar?
     return paneItem instanceof TextEditor
 
-  bufferChanged: ({newRange}) =>
-    @changeUpdateRange ?=
-      start: newRange.start.row
-      end: newRange.end.row
+  # Buffer change handling. The goal is to remove symbols from the `symbolList`
```

Comment:
@nathansobo I know you're busy, but getting your read on this would be helpful. The change management for this symbol provider needs to be a more complicated than the FuzzyProvider for :racehorse:. This comment hopefully explains it.